### PR TITLE
feat: add generic coverage script and codecov provider

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
## Jira ID

[XX-XX]

### CODE REVIEW

**Comment by jaredallard** [comment]
PR description pls

File: shell/ci/testing/codecov/get-upload-token.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  echo "Script expects the owner/organization and name of repository passed to it as parameters." >&2
+  exit 1
+fi
+
+owner="$1"
+repo="$2"
+
+info="$(curl -X GET "https://codecov.io/api/gh/$owner/$repo" -H "Authorization: $CODECOV_API_KEY")"
+
+# shellcheck disable=SC2181
```
Comment (jaredallard) [inline shell/ci/testing/codecov/get-upload-token.sh:18]: nit: why

File: shell/ci/testing/codecov/get-upload-token.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
```
Comment (jaredallard) [inline shell/ci/testing/codecov/get-upload-token.sh:1]: nit: short description of this shell script below the she-bang

**Review by jaredallard — APPROVED** [review]

File: shell/ci/testing/codecov/get-upload-token.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+if [[ -z $CODECOV_API_KEY ]]; then
+  echo "CODECOV_API_KEY environment variable is empty." >&2
+  exit 1
+fi
+
+if [[ $# -ne 2 ]]; then
+  echo "Script expects the owner/organization and name of repository passed to it as parameters." >&2
```
Comment (artemchernyak-outreach) [inline shell/ci/testing/codecov/get-upload-token.sh:9]: nit: might be nice to add a more standard `USAGE:` style line here.

File: shell/ci/testing/codecov/upload-coverage.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  echo "CODECOV_API_KEY environment variable is empty." >&2
+  exit 1
+fi
+
+if [[ $CIRCLECI == "true" ]]; then
+  owner="${CIRCLE_PROJECT_USERNAME}"
+  repo="${CIRCLE_PR_REPONAME:-$CIRCLE_PROJECT_REPONAME}"
+else
+  if [[ $# -lt 2 ]]; then
+    echo "When running outside of CIRCLECI the first two positional arguments passed to this script need to be owner and repository." >&2
```
Comment (artemchernyak-outreach) [inline shell/ci/testing/codecov/upload-coverage.sh:15]: nit: same here about the `USAGE:` line.

**Comment by jaredallard** [comment]
Still need to update the CircleCI contexts to include `circleci-credentials` now.

**Comment by jaredallard** [comment]
> Still need to update the CircleCI contexts to include `circleci-credentials` now.

PR for that: https://github.com/getoutreach/box/pull/12

**Pull request closed without merging**